### ISSUE TITLE
Only running karma test, dojo-webpack-plugin@3.0.1 DojoAMDMainTemplate.runtime.js::resolveTernaryHasExpression() throws "Module not found: dojo/has" error 

### ISSUE THREAD

**Issue by g2nconnell** [OpenNTF/dojo-webpack-plugin#342]
When running karma/jasmine tests (actual app works fine -- only tests fail),

dojo-webpack-plugin@3.0.1 DojoAMDMainTemplate.runtime.js::resolveTernaryHasExpression() throws "Module not found: dojo/has" error 
for any source file that requires "dojo/on", or anything that includes "dojo/on", which does this: 
on.js:
  define(["./has!dom-addeventlistener?:./aspect", "./_base/kernel", "./sniff"], function(aspect, dojo, has)
The reason seems to be that req.absMids is empty inside findModule when it is called by resolveTernaryHasExpression.

dojo-webpack-plugin@2.8.19  works fine with the same sources.

dojo-webpack-plugin 2.8.19

runtime.js

 
        (this || window)["webpackJsonp"].registerAbsMids = registerAbsMids;
 

commons.js

    if (jsonpArray.registerAbsMids) {
        jsonpArray.registerAbsMids(absMids);   // IT DOES THIS
    } else {
        var absMidsWaiting = jsonpArray.absMidsWaiting = jsonpArray.absMidsWaiting || [];
        absMidsWaiting.push(absMids);
    }
}

function() {
    var absMids = {
        "cloudCenterRouter": "./assets/js/cloudCenterRouter.js",
        "dao/abstractDao": "./assets/js/dao/abstractDao.js",
        "dao/mockDao": "./assets/js/dao/mockDao.js",
        "config": "./config/default-dev/config.js",
        // "/Users/gconnell/github/cloudCenter/web-client-application/node_modules/dojo-webpack-plugin/lib/NoModule.js" = "./node_modules/dojo-webpack-plugin/lib/NoModule.js"
        // "./../../../../../../node_modules/webpack/buildin/global.js" = "./node_modules/webpack/buildin/global.js"
        "backbone": "./target/debug/2P/derived/ui/Backbone/backbone.js",
        "underscore": "./target/debug/2P/derived/ui/Underscore/underscore.min.js",
        "dojo/_base/config": "./target/debug/2P/derived/ui/dojo/_base/config.js",
        "dojo/_base/kernel": "./target/debug/2P/derived/ui/dojo/_base/kernel.js",
        "dojo/aspect": "./target/debug/2P/derived/ui/dojo/aspect.js",
        "dojo/has": "./target/debug/2P/derived/ui/dojo/has.js",
        "dojo/on": "./target/debug/2P/derived/ui/dojo/on.js",
        "dojo/sniff": "./target/debug/2P/derived/ui/dojo/sniff.js",
        "jquery": "./target/debug/2P/derived/ui/jQuery/jquery-2.2.0.min.js"// "/Users/gconnell/github/cloudCenter/web-client-application/test/spec/CloudCenterRouterSpec.js" = "./test/spec/CloudCenterRouterSpec.js"
    };
    var globalObj = this || window;
    var jsonpArray = globalObj["webpackJsonp"] = globalObj["webpackJsonp"] || [];
    if (jsonpArray.registerAbsMids) {
        jsonpArray.registerAbsMids(absMids);  // IT DOES THIS
    } else {
        var absMidsWaiting = jsonpArray.absMidsWaiting = jsonpArray.absMidsWaiting || [];
        absMidsWaiting.push(absMids);
    }
}
)()

 resolveTernaryHasExpression(expr)  called with expr = "dom-addeventlistener?:./target/debug/2P/derived/ui/dojo/aspect.js"

findModule(mid, referenceModule, noInstall, asModuleObj) call with mid = "dojo/has", referenceModule = null, noInstall = false, asModuleObj = undefined.

Inside findModule, 
if (mid in req.absMids && __webpack_require__.m[req.absMids[mid]])  is TRUE because req.absMids = {
backbone: "./target/debug/2P/derived/ui/Backbone/backbone.js"
cloudCenterRouter: "./assets/js/cloudCenterRouter.js"
config: "./config/default-dev/config.js"
dao/abstractDao: "./assets/js/dao/abstractDao.js"
dao/mockDao: "./assets/js/dao/mockDao.js"
dojo/_base/config: "./target/debug/2P/derived/ui/dojo/_base/config.js"
dojo/_base/kernel: "./target/debug/2P/derived/ui/dojo/_base/kernel.js"
dojo/aspect: "./target/debug/2P/derived/ui/dojo/aspect.js"
dojo/has: "./target/debug/2P/derived/ui/dojo/has.js"
dojo/on: "./target/debug/2P/derived/ui/dojo/on.js"
dojo/sniff: "./target/debug/2P/derived/ui/dojo/sniff.js"
jquery: "./target/debug/2P/derived/ui/jQuery/jquery-2.2.0.min.js"
underscore: "./target/debug/2P/derived/ui/Underscore/underscore.min.js"
__proto__: Object
}

It finds "dojo/has" and all works.

dojo-webpack-plugin 3.0.1

commons.js

(function() {
    var absMids = {
        "cloudCenterRouter": 4,
        "dao/abstractDao": 14,
        "dao/mockDao": 13,
        "config": 15,
        // "C:\\Users\\Glen\\github\\cloudcenter\\web-client-application\\node_modules\\dojo-webpack-plugin\\lib\\NoModule.js" = 8
        "backbone": 5,
        "underscore": 1,
        "dojo/_base/config": 10,
        "dojo/_base/kernel": 9,
        "dojo/aspect": 12,
        "dojo/has": 6,
        "dojo/on": 7,
        "dojo/sniff": 11,
        "jquery": 2 // "C:/Users/Glen/github/cloudcenter/web-client-application/test/spec/CloudCenterRouterSpec.js" = 0
    };
    var globalObj = this || window;
    var jsonpArray = globalObj["webpackChunkcloudcenter2_webapp"] = globalObj["webpackChunkcloudcenter2_webapp"] || [];
    if (jsonpArray.registerAbsMids) {
        jsonpArray.registerAbsMids(absMids);
    } else {
        var absMidsWaiting = jsonpArray.absMidsWaiting = jsonpArray.absMidsWaiting || []; // IT DOES THIS because registerAbsMids never assigned to jsonpArray
        absMidsWaiting.push(absMids);
    }
} 
 
runtime.js

resolveTernaryHasExpression(expr)  called with expr = "dom-addeventlistener?:12"

findModule(mid, referenceModule, noInstall, asModuleObj) call with mid = "dojo/has", referenceModule = null, noInstall = false, asModuleObj = undefined.
Inside fineModule, 
if (mid in req.absMids && __webpack_require__.m[req.absMids[mid]])  is FALSE because req.absMids = {__proto__: Object}  It's empty so, get error "Module not found: dojo/has".

**Comment by chuckdumont** [comment]
@g2nconnell - can you provide a test case that reproduces the problem.  I can't reproduce it.  I'm using Istanbul/jasmine instead of karma/jasmine, but dojo/on works fine.

**Comment by g2nconnell** [comment]
Here's a compressed file with all but the minimum stripped out.  Refer to README.txt.
[bugrepro.zip](https://github.com/OpenNTF/dojo-webpack-plugin/files/6917688/bugrepro.zip)

**Comment by chuckdumont** [comment]
When I run `npm install` I get the following error:
```
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/karma-mw-browser-support - Not found
npm ERR! 404
npm ERR! 404  'karma-mw-browser-support@0.0.1-pre.2021.2.24.19.15.3' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404 It was specified as a dependency of 'bugrepro'
```
It seems that you're loading this from a local artifactory repo

**Comment by g2nconnell** [comment]
Sorry, I missed stripping out one thing.  Just remove that line from package.json

**Comment by g2nconnell** [comment]
I removed that dependency.  Here's a new repro test project
[bugrepro.zip](https://github.com/OpenNTF/dojo-webpack-plugin/files/6917947/bugrepro.zip)

**Comment by chuckdumont** [comment]
Doesn't seem to like running without it:
```
Starting Unit Tests...
02 08 2021 11:25:25.683:ERROR [plugin]: Cannot find plugin "karma-mw-browser-support".
  Did you forget to install it?
  npm install karma-mw-browser-support --save-dev

START:
Webpack bundling...

02 08 2021 11:25:27.569:ERROR [karma-server]: Error: Found 1 load error
    at Server.<anonymous> (C:\Users\Chuck\git\bugrepro\node_modules\karma\lib\server.js:239:26)
    at Object.onceWrapper (events.js:421:28)
    at Server.emit (events.js:327:22)
    at emitListeningNT (net.js:1352:10)
    at processTicksAndRejections (internal/process/task_queues.js:79:21)
```

**Comment by g2nconnell** [comment]
I removed that dependency in the latest zip file (from karma.conf.js).  Please try with the latest .zip.

**Comment by chuckdumont** [comment]
OK, I can reproduce with the latest zip

**Comment by g2nconnell** [comment]
Great! Thanks for your help.

**Comment by chuckdumont** [comment]
Fixed in 3.0.2
